python/ray/serve/_private/application_state.py

```diff
@@ -423,18 +424,28 @@ def _determine_app_status(self) -> Tuple[ApplicationStatus, str]:
         if self._target_state.deleting:
             return ApplicationStatus.DELETING, ""
 
-        num_healthy_deployments = 0
+        num_healthy_or_autoscaling_deployments = 0
+        num_updating_deployments = 0
         unhealthy_deployment_names = []
 
         for deployment_status in self.get_deployments_statuses():
             if deployment_status.status == DeploymentStatus.UNHEALTHY:
                 unhealthy_deployment_names.append(deployment_status.name)
-            if deployment_status.status == DeploymentStatus.HEALTHY:
-                num_healthy_deployments += 1
+            elif (
+                deployment_status.status == DeploymentStatus.HEALTHY
+                or deployment_status.status_trigger
+                == DeploymentStatusTrigger.AUTOSCALING
+            ):
+                # If all deployments are HEALTHY or autoscaling, then
+                # application is RUNNING
+                num_healthy_or_autoscaling_deployments += 1
```

Comment:
Let's avoid mixing up healthy and autoscaling deployments, and instead count them separately (keeps code cleaner and easier to reason about)